hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/IncrementalBlockReportManager.java

```diff
@@ -212,6 +212,8 @@ void sendIBRs(DatanodeProtocol namenode, DatanodeRegistration registration,
       namenode.blockReceivedAndDeleted(registration, bpid, reports);
       success = true;
     } finally {
+      /* set blocks to zero */
+      this.dnMetrics.resetBlocksInPendingIBR();
```

Comment:
Couldn't check in detail but shouldn't this be in success block? In case it isn't success, it puts the blocks back into the queue